File: firstclicks-ui/src/services/models/course-public-dto.ts
/* tslint:disable */
/* eslint-disable */
import { TechStack } from '../models/tech-stack';
import { TutorProfilePublic } from '../models/tutor-profile-public';
export interface CoursePublicDto {
  coverPath?: string;
  createdDate?: string;
  description?: string;
  id?: number;
  level?: 'ENTRY' | 'INTERMEDIATE' | 'ADVANCE';
  name?: string;
  studentReview?: Array<string>;
  studentStars?: number;
  techStack?: Array<TechStack>;
  tutor?: TutorProfilePublic;
  updatedDate?: string;
}

File: firstclicks-server/src/main/java/com/project/firstclicks/dto/CoursePublicDTO.java
package com.project.firstclicks.dto;

import java.sql.Timestamp;
import java.time.LocalDateTime;

import java.util.Set;
import java.util.List;

import com.project.firstclicks.entity.Course;
import com.project.firstclicks.entity.TechStack;

import lombok.Data;

@Data
public class CoursePublicDTO {
	private Integer id;
	private String name;
	private String description;
	private String coverPath;

	private Set<TechStack> techStack;
	private Course.Level level;
	private TutorProfilePublic tutor;
	private Timestamp createdDate;
	private Timestamp updatedDate;

	
	
	private List<String> studentReview;
	private Integer studentStars;
}
